AuditRulesMonitor.cpp

```diff
@@ -0,0 +1,228 @@
+//
+// Created by tad on 3/18/19.
+//
+
+#include "AuditRulesMonitor.h"
+
+#include "Logger.h"
+#include "ExecUtil.h"
+
+void AuditRulesMonitor::run() {
+    Logger::Info("AuditRulesMonitor: Starting");
+    while(!_sleep(1000)) {
+        auto now = std::chrono::steady_clock::now();
+
+        if (now - _last_auoms_file_check > std::chrono::seconds(60)) {
+            _last_auoms_file_check = now;
+            get_desired_rules();
+        }
+
+        if (now - _last_audit_file_check > std::chrono::seconds(5*60)) {
+            _last_audit_file_check = now;
+            check_file_rules();
+        }
+
+        if (!check_kernel_rules()) {
+            Logger::Error("AuditRulesMonitor: Encountered unrecoverable error, stopping");
+            return;
+        }
+    }
+}
+
+void AuditRulesMonitor::on_stop() {
+
+}
+
+void AuditRulesMonitor::get_desired_rules() {
+    try {
+        auto rules = ReadAuditRulesFromDir(_audit_rules_dir);
+        _desired_rules.resize(0);
+        for (auto& rule: rules) {
+            // Only include the rule in the desired rules if it is supported on the host system
+            if (rule.IsSupported()) {
```

Comment:
There is no easy way to distinguish the type of load failure. Also, if a desired rule never loads (because it is not supported) then the code will perpetually be trying to load desired rules.